coldfront/plugins/qumulo/views/update_allocation_view.py

```diff
@@ -164,22 +167,16 @@ def set_access_users(
             allocation_user.user.username for allocation_user in allocation_users
         ]
 
-        for access_user in access_users:
-            if access_user not in allocation_usernames:
-                AclAllocations.add_user_to_access_allocation(
-                    access_user, access_allocation
-                )
-                active_directory_api.add_user_to_ad_group(
-                    access_user, access_allocation.get_attribute("storage_acl_name")
-                )
-
-        for allocation_username in allocation_usernames:
-            if allocation_username not in access_users:
-                allocation_users.get(user__username=allocation_username).delete()
-                active_directory_api.remove_user_from_group(
-                    allocation_username,
-                    access_allocation.get_attribute("storage_acl_name"),
-                )
+        users_to_add = list(set(access_users) - set(allocation_usernames))
+        async_task(addUsersToADGroup, users_to_add, access_allocation)
+
+        users_to_remove = set(allocation_usernames) - set(access_users)
+        for allocation_username in users_to_remove:
```

Comment:
I doubt we're ever going to be bulk-removing large numbers of users, so I think keeping it like this, where things are removed inline, makes sense.